checker/check_result_test.go

```diff
@@ -0,0 +1,473 @@
+			args: args{
+				name:   "",
+				reason: "",
+			},
+			want: CheckResult{
+				Name:    "",
+				Reason:  "",
+				Score:   10,
+				Version: 2,
+			},
+		},
+		{
+			name: "a reason",
+			args: args{
+				name:   "name",
+				reason: "reason",
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Score:   10,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := CreateMaxScoreResult(tt.args.name, tt.args.reason); !cmp.Equal(got, tt.want) { //nolint:govet
+				t.Errorf("CreateMaxScoreResult() = %v, want %v", got, cmp.Diff(got, tt.want))
+			}
+		})
+	}
+}
+
+func TestCreateMinScoreResult(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		name   string
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want CheckResult
+	}{
+		{
+			name: "empty",
+			args: args{
+				name:   "",
+				reason: "",
+			},
+			want: CheckResult{
+				Name:    "",
+				Reason:  "",
+				Score:   0,
+				Version: 2,
+			},
+		},
+		{
+			name: "a reason",
+			args: args{
+				name:   "name",
+				reason: "reason",
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Score:   0,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := CreateMinScoreResult(tt.args.name, tt.args.reason); !cmp.Equal(got, tt.want) {
+				t.Errorf("CreateMinScoreResult() = %v, want %v", got, cmp.Diff(got, tt.want))
+			}
+		})
+	}
+}
+
+func TestCreateInconclusiveResult(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		name   string
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want CheckResult
+	}{
+		{
+			name: "empty",
+			args: args{
+				name:   "",
+				reason: "",
+			},
+			want: CheckResult{
+				Name:    "",
+				Reason:  "",
+				Score:   -1,
+				Version: 2,
+			},
+		},
+		{
+			name: "a reason",
+			args: args{
+				name:   "name",
+				reason: "reason",
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Score:   -1,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := CreateInconclusiveResult(tt.args.name, tt.args.reason); !cmp.Equal(got, tt.want) {
+				t.Errorf("CreateInconclusiveResult() = %v, want %v", got, cmp.Diff(got, tt.want))
+			}
+		})
+	}
+}
+
+func TestCreateRuntimeErrorResult(t *testing.T) {
+	t.Parallel()
+	type args struct { //nolint:govet
+		name string
+		e    error
+	}
+	tests := []struct {
+		name string
+		args args
+		want CheckResult
+	}{
+		{
+			name: "empty",
+			args: args{
+				name: "",
+				e:    errors.New("runtime error"), //nolint:goerr113
+			},
+			want: CheckResult{
+				Name:    "",
+				Reason:  "runtime error",
+				Score:   -1,
+				Version: 2,
+				Error:   errors.New("runtime error"), //nolint:goerr113
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
please shadow tt here to avoid race conditions
```
tt := tt
```

Reply:
Resolved all of them. Not commenting on each one of them.